Lib/test/test_collections.py

```diff
@@ -1916,6 +1916,22 @@ def test_helper_function(self):
         self.assertEqual(dict(c), {'a': 5, 'b': 2, 'c': 1, 'd': 1, 'r':2 })
 
 
+################################################################################
+### OrderedDict
+################################################################################
+
+class TestOrderedDict(unittest.TestCase):
+
+    def test_recursive_repr(self):
+        d = OrderedDict()
+        d[42] = d.values()
+        r = repr(d)
+        self.assertEqual(r, 'OrderedDict([(42, odict_values([...]))])')
```

Comment:
The concrete implementation is implementation dependent. For example in PyPy3 it is `OrderedDict([(42, _OrderedDictValuesView(...))])`. I think it enough to test that repr() doesn't raise an exception in other implementations.
```python
r = repr(d)
if test.support.check_impl_detail(cpython=True):
    self.assertEqual(r, 'OrderedDict([(42, odict_values([...]))])')
```